src/screens/AirdropDashboard/components/AirdropInProgress.tsx

```diff
@@ -1,47 +1,95 @@
-  { balance: 13, name: 'Wallet name B', address: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' },
-];
+export const AirdropInProgress: FC<Props> = props => {
+  let _carouselRef: Carousel<AirdropWalletCardData>;
 
-// TODO: ultimately should come from API / Redux store
-const registeredWallets = [
-  { balance: 26, name: 'Wallet name C', address: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' },
-  { balance: 8, name: 'Wallet name D', address: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' },
-];
+  const setRef = (carouselRef: Carousel<AirdropWalletCardData>) => {
+    _carouselRef = carouselRef;
+  };
 
-export const AirdropInProgress: FC<Props> = ({ navigation }) => {
-  let _carouselRef: Carousel<AirdropWalletDetails>;
+  const AirdropInProgressContent: FC<Props> = ({
+    error,
+    loading,
+    subscribedWallets,
+    availableWallets,
+    subscribeWallet,
+  }) => {
+    if (loading) {
+      return (
+        <View style={styles.loadingContainer}>
+          <Loading />
+        </View>
+      );
+    }
+    if (error) {
+      return (
+        <View style={styles.errorContainer}>
+          <Error />
+        </View>
+      );
+    }
 
-  const setRef = (carouselRef: Carousel<AirdropWalletDetails>) => {
-    _carouselRef = carouselRef;
+    return (
+      <>
+        {subscribedWallets?.length > 0 ? (
+          <>
+            <AirdropWalletsCarousel styles={styles.carouselStyles} items={subscribedWallets} setRef={setRef} />
+            <View style={styles.walletsListContainer}>
+              <AirdropWalletsList
+                wallets={subscribedWallets}
+                title={i18n.airdrop.dashboard.registeredWallets}
+                itemCallToAction={(wallet: Wallet) => (
+                  <RegisteredWalletAction onActionClick={() => setCarouselActiveElement(wallet)} />
+                )}
+              />
+            </View>
+          </>
+        ) : (
+          <>
+            <Image source={images.airdrop} style={styles.airdropImage} />
+            <Text style={styles.description}>{i18n.airdrop.dashboard.desc2}</Text>
+          </>
+        )}
+        {availableWallets?.length > 0 && (
+          <View style={styles.walletsListContainer}>
+            <AirdropWalletsList
+              wallets={availableWallets}
+              title={i18n.airdrop.dashboard.availableWallets}
+              itemCallToAction={(wallet: Wallet) => (
+                <AvailableWalletAction onActionClick={() => subscribeWallet(wallet)} />
+              )}
+            />
+          </View>
+        )}
+      </>
+    );
   };
 
-  const setCarouselActiveElement = (wallet: AirdropWalletDetails) => {
-    const snapIndex = registeredWallets.findIndex(w => w.name === wallet.name && w.balance === wallet.balance);
+  const setCarouselActiveElement = (wallet: Wallet) => {
+    const snapIndex = props.subscribedWallets.findIndex(w => w.label === wallet.label && w.balance === wallet.balance);
```

Comment:
I know that we have unique label but it seems that comparing id would be more robust solution

Reply:
done